src/nvim/decoration.c

```diff
@@ -105,12 +107,13 @@ void decor_add(buf_T *buf, int row, int row2, Decoration *decor, bool hl_id)
       buf->b_virt_line_blocks++;
     }
     if (decor_has_sign(decor)) {
+      decor->sign_add_id = sign_add_id++;
       buf->b_signs++;
     }
     if (decor->sign_text) {
       buf->b_signs_with_text++;
       // TODO(lewis6991): smarter invalidation
```

Comment:
Can this be removed now that the sentinel line is being checked for extmark signs in `buf_signcols_add_check()`, or did you plan something else @lewis6991?

Reply:
Maybe? I can't really remember without deep diving. If you think it's optimal (...enough) then feel free to remove.